### PR TITLE
Refactor/cloud breach s3

### PR DESCRIPTION
**Pull request by andrew-aiken**
#### Overview of Changes
- Updated deprecated code (https://github.com/RhinoSecurityLabs/cloudgoat/pull/201)
- Formatted code `terraform fmt`
- Linting
  - Adding variable types
- Remove duplicated & redundant code
  - ACL block by default (https://github.com/RhinoSecurityLabs/cloudgoat/issues/198)
    - Similar change to (https://github.com/RhinoSecurityLabs/cloudgoat/pull/193)
- PR template to repo

#### Testing
Tested with version `1.2.0` & `1.5.6` of Terraform

### CODE REVIEW

File: scenarios/cloud_breach_s3/terraform/data_sources.tf [hunk 1]
```diff
@@ -1,8 +1,2 @@
 #AWS Account Id
-data "aws_caller_identity" "aws-account-id" {
-
-}
-#S3 Full Access Policy
-data "aws_iam_policy" "s3-full-access" {
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/data_sources.tf:6]: Redundant

File: scenarios/cloud_breach_s3/terraform/ec2.tf [hunk 2]
```diff
@@ -1,122 +1,129 @@
...
     {
-      "Action": "sts:AssumeRole",
-      "Principal": {
-        "Service": "ec2.amazonaws.com"
-      },
-      "Effect": "Allow",
-      "Sid": ""
+      Version = "2012-10-17",
+      Statement = [
+        {
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/ec2.tf:8]: jsonencode

File: scenarios/cloud_breach_s3/terraform/ec2.tf [hunk 3]
```diff
@@ -1,122 +1,129 @@
...
-EOF
-  tags = {
-      Name = "cg-banking-WAF-Role-${var.cgid}"
-      Stack = "${var.stack-name}"
-      Scenario = "${var.scenario-name}"
-  }
-}
 
-#IAM Role Policy Attachment
-resource "aws_iam_role_policy_attachment" "cg-banking-WAF-Role-policy-attachment-s3" {
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/ec2.tf:27]: Redundant

File: scenarios/cloud_breach_s3/terraform/ec2.tf [hunk 4]
```diff
@@ -1,122 +1,129 @@
...
   ingress {
-      from_port = 22
-      to_port = 22
-      protocol = "tcp"
-      cidr_blocks = var.cg_whitelist
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = var.cg_whitelist
   }
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/ec2.tf:47]: formatting

File: scenarios/cloud_breach_s3/terraform/ec2.tf [hunk 5]
```diff
@@ -1,122 +1,129 @@
...
-        private_key = "${file(var.ssh-private-key-for-ec2)}"
-        host = self.public_ip
-      }
+  ami                         = "ami-0a313d6098716f372"
+  instance_type               = "t2.micro"
+  iam_instance_profile        = aws_iam_instance_profile.cg-ec2-instance-profile.name
+  subnet_id                   = aws_subnet.cg-public-subnet-1.id
+  associate_public_ip_address = true
+
+  vpc_security_group_ids = [
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/ec2.tf:103]: Formatting

File: scenarios/cloud_breach_s3/terraform/locals.tf [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    "cardholder_data_primary.csv",
+    "cardholder_data_secondary.csv",
+    "cardholders_corporate.csv",
+    "goat.png"
+  ]
+
+  default_tags = {
+    Stack    = var.stack-name
+    Scenario = var.scenario-name
+  }
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/locals.tf:14]: Locals to combine default tags & other variables

File: scenarios/cloud_breach_s3/terraform/s3.tf [hunk 7]
```diff
@@ -1,63 +1,23 @@
...
-      Description = "CloudGoat ${var.cgid} S3 Bucket used for storing sensitive cardholder data."
-      Stack = "${var.stack-name}"
-      Scenario = "${var.scenario-name}"
-  }
-}
-resource "aws_s3_bucket_object" "cardholder-data-primary" {
-  bucket = "${aws_s3_bucket.cg-cardholder-data-bucket.id}"
-  key = "cardholder_data_primary.csv"
-  source = "../assets/cardholder_data_primary.csv"
-  tags = {
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/s3.tf:23]: Update `aws_s3_bucket_object` -> `aws_s3_object`

Combined all objects into a single looped resources

File: scenarios/cloud_breach_s3/terraform/variables.tf [hunk 8]
```diff
@@ -1,32 +1,53 @@
 #Required: AWS Profile
 variable "profile" {
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/variables.tf:2]: Add type and description

File: scenarios/cloud_breach_s3/terraform/versions.tf [hunk 9]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 1.2.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+}
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/versions.tf:10]: Add minimum TF versions (linting)

File: scenarios/cloud_breach_s3/terraform/vpc.tf [hunk 10]
```diff
@@ -1,61 +1,63 @@
 resource "aws_vpc" "cg-vpc" {
```
Comment (andrew-aiken) [inline scenarios/cloud_breach_s3/terraform/vpc.tf:1]: formatting

**Review by andrew-aiken — COMMENTED** [review]
Added some comments to the PR to show changes

File: scenarios/cloud_breach_s3/terraform/variables.tf [hunk 11]
```diff
@@ -1,32 +1,53 @@
...
-
+  description = "CGID variable for unique naming."
+  type        = string
 }
+
 #Required: User's Public IP Address(es)
 variable "cg_whitelist" {
-  default = "../whitelist.txt"
+  description = "User's public IP address, pulled from the file ../whitelist.txt"
+  default     = ["0.0.0.0/0"]
```
Comment (jdearmas) [inline scenarios/cloud_breach_s3/terraform/variables.tf:23]: All IPs by default?

Comment (andrew-aiken) [reply scenarios/cloud_breach_s3/terraform/variables.tf:23]: We could switch this to something else (`0.0.0.0/32`) Its more to switch to a default value that would be [valid](https://github.com/RhinoSecurityLabs/cloudgoat/pull/214/files/8c62eaae8bde1a01f5d2d0b48d1dd43455dfce75#diff-4bc912b2516abb43f424509cf93dad506c7da00b4d39dc0c9fa26c19309c7cf5R46) (`list(string)`)

**Review by jdearmas — APPROVED** [review]

**Pull request merged**